ksonnet-gen/ksonnet/emit.go

```diff
@@ -423,7 +424,7 @@ func (ao *apiObject) emit(m *indentWriter) {
 	for _, pm := range ao.properties.sortAndFilterBlacklisted() {
 		// Skip special properties and fields that `$ref` another API
 		// object type, since those will go in the `mixin` namespace.
-		if isSpecialProperty(pm.name) || pm.ref != nil {
+		if isSpecialProperty(pm.name) || isMixinRef(pm.ref) {
```

Comment:
Just a couple of clarifying questions for my own edification:

* what is a special property?
* where does the term mixin come from? Is this a name you assigned to the idea that many resources in k8s reference the same object type?